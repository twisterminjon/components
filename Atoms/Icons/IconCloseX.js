import React, { Component } from 'react'
import PropTypes from 'prop-types'

export default class IconDeclineCall extends Component {
  static propTypes = {
    /** Color for the icon */
    color: PropTypes.string,
    /** Size for icon in pixels */
    size: PropTypes.number,
  }
  static defaultProps = {
    color: 'rgba(0,0,0,0.75)',
    size: 26,
  }

  render() {
    const { color, size, className, style } = this.props

    const wrapperStyle = Object.assign({}, style)
    const wrapperClass = className

    return (
      <svg
        style={wrapperStyle}
        className={wrapperClass}
        xmlns="http://www.w3.org/2000/svg"
        width={size}
        height={size}
        viewBox="0 0 26 26"
        fillRule="evenodd">
        <g fill={color}>
          <polygon
            isolation="isolate"
            points="23.8 25.6 12.6 14.3 1.9 25 0.1 23.2 10.8 12.6 0 1.8 1.8 0 12.6 10.8 23.4 0 25.1 1.8 14.3 12.6 25.6 23.8 23.8 25.6"
          />
        </g>
      </svg>
    )
  }
}
