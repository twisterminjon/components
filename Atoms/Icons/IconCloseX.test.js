import React from 'react'
import IconCloseX from './IconCloseX'

describe('IconCloseX', () => {
  it('matches the snapshot', () => {
    const wrapper = window.shallow(<IconCloseX color="red" size={36} />)

    expect(wrapper).toMatchSnapshot()
  })
})
