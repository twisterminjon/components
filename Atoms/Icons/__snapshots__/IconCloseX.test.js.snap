// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`IconCloseX matches the snapshot 1`] = `
<svg
  fillRule="evenodd"
  height={36}
  style={Object {}}
  viewBox="0 0 26 26"
  width={36}
  xmlns="http://www.w3.org/2000/svg"
>
  <g
    fill="red"
  >
    <polygon
      isolation="isolate"
      points="23.8 25.6 12.6 14.3 1.9 25 0.1 23.2 10.8 12.6 0 1.8 1.8 0 12.6 10.8 23.4 0 25.1 1.8 14.3 12.6 25.6 23.8 23.8 25.6"
    />
  </g>
</svg>
`;
